kale/static_analysis/dependencies.py

```diff
@@ -61,25 +61,37 @@ def pyflakes_report(code):
     return undef_vars
 
 
-def detect_in_dependencies(nb_graph: nx.DiGraph, ignore_symbols: set = None):
+def detect_in_dependencies(nb_graph: nx.DiGraph,
+                           pipeline_parameters: dict = None):
     """Detect missing names from the code blocks in the graph.
 
     Args:
         nb_graph: nx DiGraph with pipeline code blocks
-        ignore_symbols: names to be ignored from the report
+        pipeline_parameters: Pipeline parameters dict
     """
     block_names = nb_graph.nodes()
     for block in block_names:
         source_code = '\n'.join(nb_graph.nodes(data=True)[block]['source'])
         commented_source_code = utils.comment_magic_commands(source_code)
         ins = pyflakes_report(code=commented_source_code)
 
-        if ignore_symbols:
-            ins.difference_update(set(ignore_symbols))
-        nx.set_node_attributes(nb_graph, {block: {'ins': sorted(ins)}})
-
-
-def detect_out_dependencies(nb_graph: nx.DiGraph, ignore_symbols: set = None):
+        # Pipeline parameters will be part of the names that are missing,
+        # but of course we don't want to marshal them in as they will be
+        # present as parameters
+        relevant_parameters = []
+        if pipeline_parameters:
+            # Not all pipeline parameters are needed in every pipeline step,
+            # these are the parameters that are actually needed by this step.
+            params = set(pipeline_parameters.keys())
+            relevant_parameters = ins.intersection(set(params))
```

Comment:
`intersection()` method accepts any iterable, i.e., even lists.
So this could be simplified as:
```suggestion
            relevant_parameters = ins.intersection(pipeline_parameter.keys())
```